extend/common-interface/development-branches.md

```diff
@@ -0,0 +1,53 @@
+
+### How do you tell component is executed in branch context?
+
+When [runner](/extend/docker-runner/) executes a job in a branch, it sets [`KBC_BRANCHID` environment variable](/extend/common-interface/environment/#environment-variables) to current branch id. This ID is unique for each branch in the whole stack. 
+
+The fact that the component is executed in a branch is not very important for the component itself. It behaves the same way and all the heavy lifting is done by Keboola Connection job runner. 
+
+The only exception is if the component interacts with storage API directly using forwarded storage token. In that case it needs to take the branch ID into consideration. Any such component is a subject of a separate component review by Keboola to ensure the implementation is correct.  
+
+### Input and output mapping in development branch
+
+#### Write
+
+When a component in development branch writes data to storage, the bucket name stored in [input mapping](/transformations/mappings/#table-input-mapping) is dynamically changed by prefixing the branches internal ID. That means that your production bucket will not be overwritten.
+
+#### Read
+
+When a component in development branch reads data from storage, it first checks if there is a development version of the production bucket. If there is one, it uses it. If development bucket does not exist, it reads the data from production bucket. That way you don't need to re-run all your data extraction jobs in a development branch.
+
+### Configuration state in development branch
+
+The state is stored separately for the development branch. Notice though, that [state](/integrate/storage/api/configurations/#state) is not merged when a development branch is merged. 
+
+### Components interacting with external resources
+
+Special care needs to be taken so that components interacting with external resources do not affect production data when ran in development branch. 
+
+Snowflake writer is an example of such component. In production, a Snowflake writer configuration is created. It writes to a schema `PROD_SCHEMA` in the Snowflake database. Executing a job of this configuration in a development branch without any safeguards would write the branch data to the `PROD_SCHEMA`. Therefore, some component jobs are limited in the development branch. 
+
+The limitation is based on component's features. Currently, there are following features:
+
+* **dev-branch-configuration-unsafe**: Components with this feature can be run in development branch, but only after `{configuration:{runtime: {safe: true}}}` is explicitly set in the configuration. This can either be done via API or using the *Safe for run in branch* toggle in the configuration detail in the UI. This is transparent to you as a developer. The job runner checks if the component's job can or can not be executed. 
```

Comment:
tady by mělo být, že je to automaticky nastavený pro writery a a aplikace a pro ostatní typy komponent je to vyplý